controllers/reconcile.go

```diff
@@ -595,6 +613,50 @@ func (r *MicroK8sControlPlaneReconciler) scaleDownControlPlane(ctx context.Conte
 	return ctrl.Result{Requeue: true}, nil
 }
 
+func getClusterAgentClient(machines []clusterv1.Machine, delMachine clusterv1.Machine) (*clusteragent.Client, error) {
+	opts := clusteragent.Options{
+		IgnoreNodeIPs: sets.NewString(),
+		// NOTE(hue): Port is hard coded according to
+		// https://github.com/canonical/cluster-api-control-plane-provider-microk8s/blob/v0.6.10/control-plane-components.yaml#L96-L102
+		Port: "30000",
+	}
+	// NOTE(hue): We want to pick a random machine's IP to call POST /dqlite/remove on its cluster agent endpoint.
+	// This machine should preferably not be the <delMachine> itself but this is not forced by Microk8s.
+	for _, addr := range delMachine.Status.Addresses {
+		opts.IgnoreNodeIPs.Insert(addr.Address)
```

Comment:
Isn't it simpler and easier to follow if we ignored the machine directly? Something like a `deleteMachine.Name` check?

Reply:
I'm not sure if I understand this correctly. Changed some things in the following commit. Lemme sure if this is actually what you had in mind.